tests/test_dwarf.py

```diff
@@ -1868,6 +1868,66 @@ def test_language(self):
             lang=DW_LANG.BLISS,
         )
 
+    def test_struct_template(self):
+        dies = [
+            DwarfDie(
+                DW_TAG.structure_type,
+                [
+                    DwarfAttrib(DW_AT.name, DW_FORM.string, "point"),
+                    DwarfAttrib(DW_AT.byte_size, DW_FORM.data1, 8),
+                ],
+                [
+                    DwarfDie(
+                        DW_TAG.member,
+                        [
+                            DwarfAttrib(DW_AT.name, DW_FORM.string, "x"),
+                            DwarfAttrib(DW_AT.data_member_location, DW_FORM.data1, 0),
+                            DwarfAttrib(DW_AT.type, DW_FORM.ref4, 1),
+                        ],
+                    ),
+                    DwarfDie(
+                        DW_TAG.member,
+                        [
+                            DwarfAttrib(DW_AT.name, DW_FORM.string, "y"),
+                            DwarfAttrib(DW_AT.data_member_location, DW_FORM.data1, 4),
+                            DwarfAttrib(DW_AT.type, DW_FORM.ref4, 1),
+                        ],
+                    ),
+                    DwarfDie(
+                        DW_TAG.template_type_parameter,
+                        [
+                            DwarfAttrib(DW_AT.name, DW_FORM.string, "T"),
+                            DwarfAttrib(DW_AT.type, DW_FORM.ref4, 1),
+                        ],
+                    ),
+                    DwarfDie(
+                        DW_TAG.template_type_parameter,
+                        [
+                            DwarfAttrib(DW_AT.name, DW_FORM.string, "U"),
+                            DwarfAttrib(DW_AT.type, DW_FORM.ref4, 1),
+                        ],
+                    ),
+                ],
+            ),
+            int_die,
+        ]
+
+        t = struct_type(
+            "point",
+            8,
+            (
+                TypeMember(int_type("int", 4, True), "x", 0),
+                TypeMember(int_type("int", 4, True), "y", 32),
+            ),
+        )
+
+        die_type = self.type_from_dwarf(dies)
+        self.assertNotEqual(die_type, t)
+        self.assertEqual(len(die_type.template_parameters), 2)
+        self.assertEqual(die_type.template_parameters[0].name, "T")
+        self.assertEqual(die_type.template_parameters[1].name, "U")
+        self.assertEqual(die_type.template_parameters[0].type, int_type("int", 4, True))
+        self.assertEqual(die_type.template_parameters[1].type, int_type("int", 4, True))
```

Comment:
Let's add the tests after the Python binding support so you can compare against a constructed type instead of having to check each template parameter.